examples/nlp/language_modeling/conf/megatron_retro_config.yaml

```diff
@@ -1,127 +1,260 @@
-    knn_index: ???  # list of KNN map index files
-    retrieval_prefix: ???   # a singe path to retrieval data
+    # Dictionary: can override from CLI "model.data.data_prefix"={"train":[1.0, /path/to/data], "validation":/path/to/data, "test":/path/to/test}
+    # Or see example below:
+    # "model.data.data_prefix: {train:[1.0,/path/to/data], validation:[/path/to/data], test:[/path/to/test]}"
+    data_prefix: ???     # will be overrided by value from RETRO preprocessed workdir 
     index_mapping_dir: null # path to save index mapping .npy files, by default will save in the same location as data_prefix
-    data_impl: retmmap   # for retro model, this is the only allowed type
-    splits_string: 900,50,50
-    seq_length: ${model.encoder_seq_length}  # must be multiple of the chunk_size in your dataset
+    data_impl: mmap
+    splits_string: 900,50,50     # will be overrided by value from RETRO preprocessed workdir 
+    seq_length: ${model.encoder_seq_length}     # will be overrided by value from RETRO preprocessed workdir 
     skip_warmup: True
-    num_workers: 0
+    num_workers: 2
     dataloader_type: single # cyclic
-    neighbors: 2  # number of retrieved neighbors
+    reset_position_ids: False # Reset position ids after end-of-document token
+    reset_attention_mask: False # Reset attention mask after end-of-document token
+    eod_mask_loss: False # Mask loss for the end of document tokens
+    validation_drop_last: True # Set to false if the last partial validation samples is to be consumed
+    no_seqlen_plus_one_input_tokens: False # Set to True to disable fetching (sequence length + 1) input tokens, instead get (sequence length) input tokens and mask the last token
+    pad_samples_to_global_batch_size: False # Set to True if you want to pad the last partial batch with -1's to equal global batch size
+    shuffle_documents: True # Set to False to disable documents shuffling. Sample index will still be shuffled
+    exchange_indices_distributed: False # Set to True to exchange indices via torch.distributed instead of filesystem
+    retro_data:
+      retro_block_size: 10000
+      retro_chunk_length: 64
+
+
+  # Nsys profiling options
+  nsys_profile:
+    enabled: False
+    start_step: 10  # Global batch to start profiling
+    end_step: 10 # Global batch to end profiling
+    ranks: [0] # Global rank IDs to profile
+    gen_shape: False # Generate model and kernel details including input shapes
   
   optim:
     name: fused_adam
-    lr: 1e-4
-    weight_decay: 0.01 
+    lr: 2.5e-05
+    weight_decay: 0.1 
     betas: 
     - 0.9
-    - 0.98
+    - 0.95
     sched:
-      name: CosineAnnealing
-      warmup_steps: 500
-      constant_steps: 50000
-      min_lr: 1e-5
+      name: null
+      min_lr: null
+      warmup_steps: null
+      max_steps: null
+    # sched:
+    #   name: CosineAnnealing
+    #   warmup_steps: 500
+    #   constant_steps: 50000
+    #   min_lr: 2e-5
```

Comment:
no learning rate scheduler is used by default? if so maybe just remove it all together

Reply:
Fixed.
I added default values for learning rate scheduler.